sshd.c

```diff
@@ -657,7 +657,7 @@ send_autxctx_state(Authctxt *auth, int fd)
 
 	if ((m = sshbuf_new()) == NULL)
 		fatal("%s: sshbuf_new failed", __func__);
-	if ((r = sshbuf_put_cstring(m, auth->pw->pw_name)) != 0)
+	if ((r = sshbuf_put_cstring(m, auth->user)) != 0)
```

Comment:
The original issue reported in https://github.com/PowerShell/Win32-OpenSSH/issues/1252 is related to the authentication problem. The problem is caused by having the username the same as the machine name.
The changes in send_autxctx_state() are called from privsep_postauth(). This is post-authentication. 
This is contradicting.

I tried to replicate this issue by creating a username same as the machine name but the windows 10 throws error message ( net helpmsg 2253) "The user name may not be same as the computer name.".
This is not recommended because having user name as same as machine name can cause confuse NetBIOS.

Reply:
My issue is definitely post-auth, sorry for wrong problem identification (same error). Anyway, I fixed a real issue - see logs, please.